src/ui/components/IconRecommendedBadge/index.js

```diff
@@ -0,0 +1,64 @@
+/* @flow */
+import * as React from 'react';
+import makeClassName from 'classnames';
+
+import Icon from 'ui/components/Icon';
+import type { Props as IconProps } from 'ui/components/Icon';
+
+import './styles.scss';
+
+export type RecommendedBadgeSize = 'large' | 'small';
+
+type Props = {|
+  alt?: $PropertyType<IconProps, 'alt'>,
+  className?: string,
+  size: RecommendedBadgeSize,
+|};
+
+const IconRecommendedBadge = ({ className, size, ...iconProps }: Props) => {
+  let sizeClassName;
```

Comment:
How about replacing this `switch` with:

```js
  if (!['large', 'small'].includes(size)) {
    throw new Error(`Unknown size: "${size}"`);
  }
  const sizeClassName = `IconRecommendedBadge-${size}`;
```

which seems a bit less verbose.

Reply:
In addons-frontend, it's a bit tricky to learn about which class names are used by which components since they are global and can be used anywhere. Because of this, I think we should avoid generating class names with template strings.

Reply:
I think I can do this without a switch case, though. There are only two sizes so I can use `makeClassName()`